src/scripts/main.js

```diff
@@ -1 +1,80 @@
+/* eslint-disable no-shadow */
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.documentElement.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  // eslint-disable-next-line prefer-promise-reject-errors
+  window.setTimeout(() => reject('First promise was rejected'), 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const a = new Promise((resolve) => {
+    document.documentElement.addEventListener('click', () => {
+      resolve();
+    });
+  });
+
+  const b = new Promise((resolve) => {
+    document.documentElement.addEventListener('contextmenu', () => {
+      resolve();
+    });
+  });
+
+  const result = Promise.race([a, b]);
+
+  result.finally(() => resolve('Second promise was resolved'));
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  const a = new Promise((resolve) => {
+    document.documentElement.addEventListener('click', () => {
+      resolve();
+    });
+  });
+
+  const b = new Promise((resolve) => {
+    document.documentElement.addEventListener('contextmenu', () => {
+      resolve();
+    });
+  });
+
+  const result = Promise.all([a, b]);
+
+  result.finally(() => resolve('Third promise was resolved'));
```

Comment:
The use of `Promise.all` for the third promise is correct as it resolves only when both 'click' and 'contextmenu' events have occurred. However, ensure that the logic does not lead to multiple resolutions due to the event listeners.